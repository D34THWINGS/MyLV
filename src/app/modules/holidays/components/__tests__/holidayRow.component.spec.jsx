import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { HolidayRow } from '../holidayRow.component'

jest.unmock('../holidayRow.component')

describe('PeriodsList', () => {
  let props
  beforeEach(() => {
    props = {
      classes: {},
      holiday: {
        date: '1942-01-01',
        id: 'bar',
        periods: [],
        status: 'pending',
      },
      partner: {
        firstName: 'Foo',
        lastName: 'Bar',
      },
      onHolidayDelete: jest.fn(),
      onClick: jest.fn(),
    }
  })

  const getWrapper = () => shallow(<HolidayRow {...props} />)

  it('should render properly', () => {
    // When
    const wrapper = getWrapper()

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should render with menu when click menu button', () => {
    // Given
    const wrapper = getWrapper()

    // When
    wrapper.find('WithStyles(IconButton)').simulate('click', { currentTarget: 'foo', stopPropagation: jest.fn() })

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should not render menu button if disabled', () => {
    // Given
    props.disableMenu = true

    // When
    const wrapper = getWrapper()

    // Then
    expect(wrapper.find('WithStyles(IconButton)')).toHaveLength(0)
  })

  it('should not render menu button if status is not pending', () => {
    // Given
    props.holiday.status = 'rejected'

    // When
    const wrapper = getWrapper()

    // Then
    expect(wrapper.find('WithStyles(IconButton)')).toHaveLength(0)
  })

  it('should render deleted partner text if no partner prop', () => {
    // Given
    props.holiday.status = 'validated'
    props.displayPartnerName = true

    // When
    const wrapper = getWrapper()

    // Then
    expect(wrapper).toMatchSnapshot()
  })
})
