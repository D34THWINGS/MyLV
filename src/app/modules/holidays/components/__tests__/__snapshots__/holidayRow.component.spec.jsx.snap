// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeriodsList should render properly 1`] = `
<WithStyles(TableRow)
  hover={true}
  onClick={[MockFunction]}
>
  <WithStyles(TableCell)>
    <WithStyles(IconButton)
      key="menu-btn"
      onClick={[Function]}
    >
      <pure(MoreVert) />
    </WithStyles(IconButton)>
    <WithStyles(Menu)
      anchorEl={null}
      id="holiday-menu-bar"
      key="menu-dropdown"
      onClose={[Function]}
      open={false}
    >
      <WithStyles(MenuItem)
        component={[Function]}
        onClick={[Function]}
        to="/holidays/bar/edit"
      >
        Éditer
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        onClick={[Function]}
      >
        Supprimer
      </WithStyles(MenuItem)>
    </WithStyles(Menu)>
  </WithStyles(TableCell)>
  <WithStyles(TableCell)>
    01-01-1942
  </WithStyles(TableCell)>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={true}
    mdUp={false}
    smDown={false}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(TableCell)
      key="paidHolidays"
      numeric={true}
    />
    <WithStyles(TableCell)
      key="unpaidHolidays"
      numeric={true}
    />
    <WithStyles(TableCell)
      key="conventionalHolidays"
      numeric={true}
    />
    <WithStyles(TableCell)>
      <WithStyles(Tooltip)
        placement="bottom"
        title="En attente de validation"
      >
        <pure(HourglassEmpty) />
      </WithStyles(Tooltip)>
    </WithStyles(TableCell)>
  </Hidden>
</WithStyles(TableRow)>
`;

exports[`PeriodsList should render with menu when click menu button 1`] = `
<WithStyles(TableRow)
  hover={true}
  onClick={[MockFunction]}
>
  <WithStyles(TableCell)>
    <WithStyles(IconButton)
      key="menu-btn"
      onClick={[Function]}
    >
      <pure(MoreVert) />
    </WithStyles(IconButton)>
    <WithStyles(Menu)
      anchorEl="foo"
      id="holiday-menu-bar"
      key="menu-dropdown"
      onClose={[Function]}
      open={true}
    >
      <WithStyles(MenuItem)
        component={[Function]}
        onClick={[Function]}
        to="/holidays/bar/edit"
      >
        Éditer
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        onClick={[Function]}
      >
        Supprimer
      </WithStyles(MenuItem)>
    </WithStyles(Menu)>
  </WithStyles(TableCell)>
  <WithStyles(TableCell)>
    01-01-1942
  </WithStyles(TableCell)>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={true}
    mdUp={false}
    smDown={false}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(TableCell)
      key="paidHolidays"
      numeric={true}
    />
    <WithStyles(TableCell)
      key="unpaidHolidays"
      numeric={true}
    />
    <WithStyles(TableCell)
      key="conventionalHolidays"
      numeric={true}
    />
    <WithStyles(TableCell)>
      <WithStyles(Tooltip)
        placement="bottom"
        title="En attente de validation"
      >
        <pure(HourglassEmpty) />
      </WithStyles(Tooltip)>
    </WithStyles(TableCell)>
  </Hidden>
</WithStyles(TableRow)>
`;

exports[`PeriodsList should render with partner name if displayPartnerName is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HolidayRow
    classes={Object {}}
    disableMenu={false}
    displayPartnerName={true}
    holiday={
      Object {
        "date": "1942-01-01",
        "id": "bar",
        "partner": Object {
          "firstName": "hello",
          "lastName": "world",
        },
        "periods": Array [],
        "status": "validated",
      }
    }
    onClick={[MockFunction]}
    onHolidayDelete={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(TableCell)
          className={undefined}
        >
          hello world
        </WithStyles(TableCell)>,
        <WithStyles(TableCell)
          className={undefined}
        >
          01-01-1942
        </WithStyles(TableCell)>,
        <Hidden
          implementation="js"
          lgDown={false}
          lgUp={false}
          mdDown={true}
          mdUp={false}
          smDown={false}
          smUp={false}
          xlDown={false}
          xlUp={false}
          xsDown={false}
          xsUp={false}
        >
          <WithStyles(TableCell)
            numeric={true}
          />
          <WithStyles(TableCell)
            numeric={true}
          />
          <WithStyles(TableCell)
            numeric={true}
          />
          <WithStyles(TableCell)>
            <WithStyles(Tooltip)
              placement="bottom"
              title="Demande acceptée"
            >
              <pure(Check) />
            </WithStyles(Tooltip)>
          </WithStyles(TableCell)>
        </Hidden>,
      ],
      "className": undefined,
      "hover": true,
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "hello world",
            undefined,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          "hello world",
          undefined,
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "01-01-1942",
          "className": undefined,
        },
        "ref": null,
        "rendered": "01-01-1942",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            Array [
              <WithStyles(TableCell)
                numeric={true}
              />,
              <WithStyles(TableCell)
                numeric={true}
              />,
              <WithStyles(TableCell)
                numeric={true}
              />,
            ],
            <WithStyles(TableCell)>
              <WithStyles(Tooltip)
                placement="bottom"
                title="Demande acceptée"
              >
                <pure(Check) />
              </WithStyles(Tooltip)>
            </WithStyles(TableCell)>,
          ],
          "implementation": "js",
          "lgDown": false,
          "lgUp": false,
          "mdDown": true,
          "mdUp": false,
          "smDown": false,
          "smUp": false,
          "xlDown": false,
          "xlUp": false,
          "xsDown": false,
          "xsUp": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "paidHolidays",
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "numeric": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "unpaidHolidays",
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "numeric": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "conventionalHolidays",
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "numeric": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Tooltip)
                placement="bottom"
                title="Demande acceptée"
              >
                <pure(Check) />
              </WithStyles(Tooltip)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <pure(Check) />,
                "placement": "bottom",
                "title": "Demande acceptée",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(TableCell)
            className={undefined}
          >
            hello world
          </WithStyles(TableCell)>,
          <WithStyles(TableCell)
            className={undefined}
          >
            01-01-1942
          </WithStyles(TableCell)>,
          <Hidden
            implementation="js"
            lgDown={false}
            lgUp={false}
            mdDown={true}
            mdUp={false}
            smDown={false}
            smUp={false}
            xlDown={false}
            xlUp={false}
            xsDown={false}
            xsUp={false}
          >
            <WithStyles(TableCell)
              numeric={true}
            />
            <WithStyles(TableCell)
              numeric={true}
            />
            <WithStyles(TableCell)
              numeric={true}
            />
            <WithStyles(TableCell)>
              <WithStyles(Tooltip)
                placement="bottom"
                title="Demande acceptée"
              >
                <pure(Check) />
              </WithStyles(Tooltip)>
            </WithStyles(TableCell)>
          </Hidden>,
        ],
        "className": undefined,
        "hover": true,
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "hello world",
              undefined,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            "hello world",
            undefined,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "01-01-1942",
            "className": undefined,
          },
          "ref": null,
          "rendered": "01-01-1942",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              Array [
                <WithStyles(TableCell)
                  numeric={true}
                />,
                <WithStyles(TableCell)
                  numeric={true}
                />,
                <WithStyles(TableCell)
                  numeric={true}
                />,
              ],
              <WithStyles(TableCell)>
                <WithStyles(Tooltip)
                  placement="bottom"
                  title="Demande acceptée"
                >
                  <pure(Check) />
                </WithStyles(Tooltip)>
              </WithStyles(TableCell)>,
            ],
            "implementation": "js",
            "lgDown": false,
            "lgUp": false,
            "mdDown": true,
            "mdUp": false,
            "smDown": false,
            "smUp": false,
            "xlDown": false,
            "xlUp": false,
            "xsDown": false,
            "xsUp": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "paidHolidays",
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "numeric": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "unpaidHolidays",
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "numeric": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "conventionalHolidays",
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "numeric": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Tooltip)
                  placement="bottom"
                  title="Demande acceptée"
                >
                  <pure(Check) />
                </WithStyles(Tooltip)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <pure(Check) />,
                  "placement": "bottom",
                  "title": "Demande acceptée",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
